Packs/Netskope/Integrations/NetskopeAPIv2/NetskopeAPIv2_test.py

```diff
@@ -0,0 +1,524 @@
+     - Ensure the function returns the expected updated URL list.
+    """
+    from NetskopeAPIv2 import get_updated_url_list
+
+    mock_response = util_load_json("url_list")
+
+    expected_output = [
+        {
+            "id": 1,
+            "name": "new_url_list",
+            "urls": ["www.google.com", "www.abc.com"],
+            "type": "exact",
+            "json_version": 2,
+            "modify_by": "modify_by",
+            "modify_type": "Edited",
+            "modify_time": "modify_time",
+            "pending": "applied",
+        }
+    ]
+
+    assert get_updated_url_list(mock_response) == expected_output
+
+
+# @pytest.mark.parametrize('args, expected_output', [
+#     ({
+#         'start_time': '01-07-2023 00:00',
+#         'end_time': None,
+#         'insertion_start_time': None,
+#         'insertion_end_time': None,
+#     }, (1688158800, 1689513420, None, None)),
+# ])
+# def test_convert_time_args_to_num(args, expected_output):
+#     """
+#     Scenario: Test the convert_time_args_to_num function.
+
+#     Given:
+#      - Command arguments containing time-related arguments.
+
+#     When:
+#      - Calling convert_time_args_to_num function with the command arguments.
+
+#     Then:
+#      - Ensure the function returns the expected updated time arguments.
+#     """
+#     from NetskopeAPIv2 import convert_time_args_to_num
+#     assert convert_time_args_to_num(args) == expected_output
+
+
+def test_convert_number_to_str_date():
+    """
+    Scenario: Test the convert_number_to_str_date function.
+
+    Given:
+     - A date number.
+
+    When:
+     - Calling convert_number_to_str_date function with the date number.
+
+    Then:
+     - Ensure the function returns the expected date string.
+    """
+    from NetskopeAPIv2 import convert_number_to_str_date
+
+    date_number = 1677756000
+    expected_output = "02-03-2023 11:20"
+
+    assert convert_number_to_str_date(date_number) == expected_output
+
+
+def test_fetch_incidents(client, requests_mock):
+    """
+    Scenario: Fetch incidents.
+    Given:
+     -XSOAR arguments.
+    When:
+     - fetch-incident called.
+
+    Then:
+     - Ensure value is correct.
+    """
+
+    from NetskopeAPIv2 import fetch_incidents
+
+    alert_mock_response = util_load_json("alerts")
+    requests_mock.get(f"{SERVER_URL}api/v2/events/data/alert", json=alert_mock_response)
+    event_mock_response = util_load_json("events")
+    requests_mock.get(
+        f"{SERVER_URL}api/v2/events/data/application", json=event_mock_response
+    )
+    requests_mock.get(f"{SERVER_URL}api/v2/events/data/page", json=event_mock_response)
+
+    fetch_incidents(
+        client,
+        {
+            "max_fetch": 10,
+            "fetch_events": True,
+            "first_fetch": "2023-01-01 15:00",
+            "max_events_fetch": 10,
+            "event_types": ["application", "page"],
+        },
+    )
+
+    assert demisto.getLastRun() == {"lastRun": "2018-10-24T14:13:20+00:00"}
```

Comment:
please test the `fetch_incidents` functionality, check it calls the right endpoints, check its return value or the call arguments to `demisto.incidents`, this method does not test it properly.